File: src/main/java/cc/envkeeper/app/service/ProductComponentService.java
/*
 * Copyright (c) 2022 Eric Thiebaut-George.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

package cc.envkeeper.app.service;

import cc.envkeeper.app.domain.ProductComponent;
import cc.envkeeper.app.repository.ProductComponentRepository;
import cc.envkeeper.app.service.dto.ProductComponentDTO;
import cc.envkeeper.app.service.mapper.ProductComponentMapper;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.Optional;

/**
 * Service Implementation for managing {@link ProductComponent}.
 */
@Service
@Transactional
public class ProductComponentService {

    private final Logger log = LoggerFactory.getLogger(ProductComponentService.class);

    private final ProductComponentRepository productComponentRepository;

    private final ProductComponentMapper productComponentMapper;

    public ProductComponentService(ProductComponentRepository productComponentRepository, ProductComponentMapper productComponentMapper) {
        this.productComponentRepository = productComponentRepository;
        this.productComponentMapper = productComponentMapper;
    }

    /**
     * Save a productComponent.
     *
     * @param productComponentDTO the entity to save.
     * @return the persisted entity.
     */
    public ProductComponentDTO save(ProductComponentDTO productComponentDTO) {
        log.debug("Request to save ProductComponent : {}", productComponentDTO);
        ProductComponent productComponent = productComponentMapper.toEntity(productComponentDTO);
        productComponent = productComponentRepository.save(productComponent);
        return productComponentMapper.toDto(productComponent);
    }

    /**
     * Get all the productComponents.
     *
     * @param pageable the pagination information.
     * @return the list of entities.
     */
    @Transactional(readOnly = true)
    public Page<ProductComponentDTO> findAll(Pageable pageable) {
        log.debug("Request to get all ProductComponents");
        return productComponentRepository.findAll(pageable)
            .map(productComponentMapper::toDto);
    }


    /**
     * Get one productComponent by id.
     *
     * @param id the id of the entity.
     * @return the entity.
     */
    @Transactional(readOnly = true)
    public Optional<ProductComponentDTO> findOne(Long id) {
        log.debug("Request to get ProductComponent : {}", id);
        return productComponentRepository.findById(id)
            .map(productComponentMapper::toDto);
    }

    /**
     * Delete the productComponent by id.
     *
     * @param id the id of the entity.
     */
    public void delete(Long id) {
        log.debug("Request to delete ProductComponent : {}", id);
        productComponentRepository.deleteById(id);
    }
}

File: src/main/webapp/app/entities/product-component/product-component.service.ts
/*
 * Copyright (c) 2022 Eric Thiebaut-George.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

import { Injectable } from '@angular/core';
import { HttpClient, HttpResponse } from '@angular/common/http';
import { Observable } from 'rxjs';

import { SERVER_API_URL } from 'app/app.constants';
import { createRequestOption } from 'app/shared/util/request-util';
import { IProductComponent } from 'app/shared/model/product-component.model';

type EntityResponseType = HttpResponse<IProductComponent>;
type EntityArrayResponseType = HttpResponse<IProductComponent[]>;

@Injectable({ providedIn: 'root' })
export class ProductComponentService {
  public resourceUrl = SERVER_API_URL + 'api/product-components';

  constructor(protected http: HttpClient) {}

  create(productComponent: IProductComponent): Observable<EntityResponseType> {
    return this.http.post<IProductComponent>(this.resourceUrl, productComponent, { observe: 'response' });
  }

  update(productComponent: IProductComponent): Observable<EntityResponseType> {
    return this.http.put<IProductComponent>(this.resourceUrl, productComponent, { observe: 'response' });
  }

  find(id: number): Observable<EntityResponseType> {
    return this.http.get<IProductComponent>(`${this.resourceUrl}/${id}`, { observe: 'response' });
  }

  query(req?: any): Observable<EntityArrayResponseType> {
    const options = createRequestOption(req);
    return this.http.get<IProductComponent[]>(this.resourceUrl, { params: options, observe: 'response' });
  }

  delete(id: number): Observable<HttpResponse<{}>> {
    return this.http.delete(`${this.resourceUrl}/${id}`, { observe: 'response' });
  }
}
